src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DPeakItem.cpp

```diff
@@ -196,20 +196,78 @@ namespace OpenMS
       painter.setPen(Qt::DashLine);
       if (!found_intersection) // no intersection with bounding box of text -> normal drawing
       {
-        painter.drawLine(peak_position_widget, position_widget);
-        painter.drawLine(peak_position_widget, position_widget);
+        painter.drawLine(peak_position_widget, peak_position_widget + QPoint(0, vertical_shift));
+        painter.drawLine(peak_position_widget + QPoint(0, vertical_shift), position_widget);
       }
       else
       {
-        painter.drawLine(peak_position_widget, *closest_ip);
-        painter.drawLine(peak_position_widget, *closest_ip);
+        QPoint vertical_end = QPoint(peak_position_widget.x(), position_widget.y() + 20);
+
+        if (vertical_end.y() < peak_position_widget.y()) // label is above peak
+        {
+          painter.drawLine(peak_position_widget, vertical_end);
+          painter.drawLine(vertical_end, position_widget);   
+        }
+        else // label is below peak
+        {
+          painter.drawLine(peak_position_widget, *closest_ip);
+          painter.drawLine(peak_position_widget, *closest_ip);       
+        }        
       }
       painter.restore();
       delete(ip);
       delete(closest_ip);
     }
 
-    painter.drawText(bounding_box_, Qt::AlignCenter, text_);
+    // some pretty printing
+    QString text = text_;
+    if (!text.contains("<\\")) // don't process HTML strings again
+    {
+      // extract ion index
+      QRegExp reg_exp("[a|b|c|x|y|z](\\d+)");
```

Comment:
according to the QRegExp documentation:
`[ABCD] will match an A or a B or a C or a D`
So you should not need the `|` bars, the `[]` brackets already imply `or` between the letters, I think.
Looks like you would also be matching strings like `|3+` right now.